lib/conjur/password_complexity.rb

```diff
@@ -0,0 +1,20 @@
+module Conjur
+  # Password complexity verifies if a password passes the minimum complexity
+  class PasswordComplexity
+    # special characters according to https://www.owasp.org/index.php/Password_special_characters
+    VALID_PASSWORD_REGEX = %r{^(?=.*?[A-Z].*[A-Z])                             # 2 uppercase letters
+      (?=.*?[a-z].*[a-z])                             # 2 lowercase letters
+      (?=.*?[0-9])                                    # 1 digit
+      (?=.*[ !"#$%&'()*+,-.\/:;<=>?@\[\\\]^_`{|}~]).  # 1 special character
+      {12,128}$}x       
+      
+      # 12-128 characters
+    def initialize(password)
+      @plain_password = password
+    end
+
+    def invalid?
+      @plain_password && (@plain_password.index("\n") || @plain_password !~ VALID_PASSWORD_REGEX)
+    end
+  end
+end
```

Comment:
This is just a `Password` or an `UnvalidatedPassword`.  Then in the code where you use this:

```ruby
unless UnvalidatedPassword.new(raw_pw).valid?
  # add some error
end
```
Alternatively, you just have a function on a module like:
```ruby
module Conjur
  module Password
    def self.valid?
# etc
```